### PR TITLE
feat(api)!: initial integration of registry1 api flavor

### PR DESCRIPTION
**Pull request by justinthelaw**
## OVERVIEW

Adds the Registry1 flavor of the LeapfrogAI API to the developer workflow and increases Helm chart and Dockerfile standard  (foreshadowing upcoming Helm and Dockerfile linting PR).

Once all other `registry1` flavored images are published to IronBank, the `e2e-registry1-weekly.yaml` will need to have access to a `uds-bundle.yaml` that points to packages created using the `registry1.dso.mil` container registry. For now, as each component comes online, we can use the `yq` commands to temporarily modify each one within the workflow:

```bash
uds zarf tools yq -i '.packages[1] |= del(.repository)' uds-bundles/latest/cpu/uds-bundle.yaml
uds zarf tools yq -i '.packages[1] |= .ref = "registry1"' uds-bundles/latest/cpu/uds-bundle.yaml
uds zarf tools yq -i '.packages[1] |= .path = "../../../packages/api"' uds-bundles/latest/cpu/uds-bundle.yaml
uds zarf tools yq -i '.metadata.version = "registry1"' uds-bundles/latest/cpu/uds-bundle.yaml
```

## BREAKING CHANGES

- LeapfrogAI API internal cluster service subdomain, and zarf-connect name, is now `leapfrogai-api` instead of `api`, following the convention of using the Helm chart name for template helpers
- LeapfrogAI API Zarf package now uses Zarf package template variable `IMAGE_VERSION` instead of `LEAPFROGAI_IMAGE_VERSION`, to align the build process with all other components/packages

## CHANGES

- Adds `FLAVOR` flag to the Makefile, defaulting to `upstream`
- Adds `registry1` Zarf package flavoring for the API
- Refactors manifest structure to align with UDS Common and Core best practices
- Slight optimizations to the LeapfrogAI API Dockerfile
- Implements and uses helper functions for Helm chart templating
- Adds documentation to mention and use the Registry1 flavor
- Adds weekly test for registry1 flavored API (to be refactored later on)
- Fixes UDS setup action usage and version, v0.14.0

### CODE REVIEW

**Comment by justinthelaw** [comment]
Depends on documentation updates from #902, and then one more update on top of that documentation.

**Comment by justinthelaw** [comment]
The UI tests will fail because of the changes to the API manifests (see the change made in this PR related to UI Helm charts).

**Review by CollectiveUnicorn — DISMISSED** [review]

File: packages/api/config.example.yaml
Comment (gphorvath) [inline packages/api/config.example.yaml:1]: Do we no longer need config.yaml for local (non-k8s) dev?

Comment (justinthelaw) [reply packages/api/config.example.yaml:1]: It is only required when running from src/leapfrogai_api, not in the packages directory. In local-k8s, the API uses kiwigrid.

Comment (gphorvath) [reply packages/api/config.example.yaml:1]: ahhh good call!

**Review by gphorvath — APPROVED** [review]

**Comment by justinthelaw** [comment]
Pending @YrrepNoj approval too! Thanks @gphorvath and @CollectiveUnicorn

**Review by CollectiveUnicorn — APPROVED** [review]

**Review by gphorvath — APPROVED** [review]

**Review by YrrepNoj — CHANGES_REQUESTED** [review]
One small change request and one pondering on if we should consider this PR a 'breaking' change.

File: packages/api/zarf.yaml [hunk 1]
```diff
@@ -2,16 +2,13 @@
 
 kind: ZarfPackageConfig
 metadata:
-  description: "LeapfrogAI"
+  description: "LeapfrogAI API"
   name: leapfrogai-api
-  version: "###ZARF_PKG_TMPL_LEAPFROGAI_IMAGE_VERSION###"
+  version: "###ZARF_PKG_TMPL_IMAGE_VERSION###"
```
[the remark this answers is not in the document]
Comment (justinthelaw) [reply packages/api/zarf.yaml:7]: I don't believe this is a breaking change from a user's perspective. This _mostly_ is internally used in the Makefile and the API was the only one that didn't use `IMAGE_VERSION` to modify the version on the fly during build.

File: packages/api/chart/values.yaml [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
-  exposeAPI: true
-  defaultEmbeddingsModel: "###ZARF_VAR_DEFAULT_EMBEDDINGS_MODEL###"
+    runAsNonRoot: true
+    capabilities:
+      drop:
+      - ALL
+  env:
+    port: 8080
+    exposeAPI: "###ZARF_VAR_EXPOSE_API###"
+    defaultEmbeddingsModel: "###ZARF_VAR_DEFAULT_EMBEDDINGS_MODEL###"
```
[the remark this answers is not in the document]
Comment (justinthelaw) [reply packages/api/chart/values.yaml:23]: Oh shoot, I meant to, and have been, removing these in other charts - I must have missed this one, will fix now. Thank you!

File: packages/api/zarf.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (YrrepNoj) [reply packages/api/zarf.yaml:7]: If the user is a Zarf package creator then this is kind of a breaking change. If they use the Make target then they the break will be abstracted away from them but people who create packages with the Zarf CLI would have to know there is new name to this variable.

At the end of the day I don't want us to do anything different with the PR, but we should make sure we include a bullet in our release notes to be good stewards to anyone who might be building this package themselves.

[the remark this answers is not in the document]
Comment (justinthelaw) [reply packages/api/zarf.yaml:7]: Okay that makes sense, I can add this as a breaking change in the PR title and in the notes

**Review by YrrepNoj — APPROVED** [review]
LGTM. Thanks for the updates.

**Review by CollectiveUnicorn — APPROVED** [review]

**Review by gphorvath — APPROVED** [review]

**Pull request merged**